safe/gui/tools/minimum_needs/needs_calculator_dialog.py

```diff
@@ -61,148 +100,57 @@ def __init__(self, parent=None):
-                add_ordered_combo_item(self.cboPolygonLayers, name, source)
-        # Now disable the run button if no suitable layers were found
-        # see #2206
-        ok_button = self.button_box.button(QtGui.QDialogButtonBox.Ok)
-        if found_flag:
-            self.cboPolygonLayers.setCurrentIndex(0)
-            ok_button.setEnabled(True)
-        else:
-            ok_button.setEnabled(False)
-
-    # prevents actions being handled twice
-    # noinspection PyPep8Naming
-    @pyqtSignature('int')
-    def on_cboPolygonLayers_currentIndexChanged(self, index):
-        """Automatic slot executed when the layer is changed to update fields.
-
-        :param index: Passed by the signal that triggers this slot.
-        :type index: int
-        """
-        layer_id = self.cboPolygonLayers.itemData(
-            index, QtCore.Qt.UserRole)
-        # noinspection PyArgumentList
-        layer = QgsMapLayerRegistry.instance().mapLayer(layer_id)
-        fields = layer.pendingFields()
-        self.cboFields.clear()
-        has_fields = False
-        for field in fields:
-            # LOGGER.info(field.typeName())
-            # TODO exclude dates too? TS
-            if field.typeName() != 'String':
-                has_fields = True
-                add_ordered_combo_item(
-                    self.cboFields, field.name(), field.name())
-
-        # Now disable the run button if no suitable fields were found
-        # see #2206
-        ok_button = self.button_box.button(QtGui.QDialogButtonBox.Ok)
-        if not has_fields:
-            ok_button.setEnabled(False)
-        else:
-            ok_button.setEnabled(True)
+        # count each minimum needs for every features
+        for needs in minimum_needs_post_processors:
+            run_single_post_processor(input_layer, needs)
 
     def accept(self):
         """Process the layer and field and generate a new layer.
 
         .. note:: This is called on OK click.
 
         """
-        index = self.cboFields.currentIndex()
-        field_name = self.cboFields.itemData(
-            index, QtCore.Qt.UserRole)
-
-        index = self.cboPolygonLayers.currentIndex()
-        layer_id = self.cboPolygonLayers.itemData(
-            index, QtCore.Qt.UserRole)
-        # noinspection PyArgumentList
-        layer = QgsMapLayerRegistry.instance().mapLayer(layer_id)
-
-        file_name = layer.source()
+        # create memory layer
+        output_layer = (
+            '%s_minimum_needs' % self.layer.currentLayer().name())
+        output_layer = create_memory_layer(
+            output_layer,
+            self.layer.currentLayer().geometryType(),
+            self.layer.currentLayer().crs(),
+            self.layer.currentLayer().fields())
+
+        # monkey patching input layer to make it work with
+        # prepare vector layer function
+        temp_layer = self.layer.currentLayer()
+        temp_layer.keywords = {'layer_purpose': 'aggregation'}
```

Comment:
`from safe.definitions.layer_purpose import layer_purpose_aggregation` and then use `layer_purpose_aggregation['key']` instead of using the string 'aggregation'